colcon_cmake/task/cmake/build.py

```diff
@@ -313,27 +311,24 @@ def _get_make_arguments(self):
     async def _install(self, args, env):
         self.progress('install')
 
-        generator = get_generator(args.build_base)
-        if 'Visual Studio' not in generator:
-            if CMAKE_EXECUTABLE is None:
-                raise RuntimeError("Could not find 'cmake' executable")
-            cmd = [
-                CMAKE_EXECUTABLE, '--build', args.build_base,
-                '--target', 'install']
+        if CMAKE_EXECUTABLE is None:
+            raise RuntimeError("Could not find 'cmake' executable")
+        cmd = [
+            CMAKE_EXECUTABLE, '--build', args.build_base,
+            '--target', 'install']
+        multi_configuration_generator = is_multi_configuration_generator(
+            args.build_base, args.cmake_args)
+        if multi_configuration_generator:
+            cmd += ['--config', self._get_configuration(args)]
+        else:
+            # Add make/ninja arguments for the number of job threads to use.
             job_args = self._get_make_arguments()
             if job_args:
+                # Add arguments for the number of job threads to use
+                # Arguably, _get_make_arguments() should be be configured to
+                # return arguments based on the current generator, not just for
+                # make (and ninja by coincidence). The '--' extension may be
+                # better returned from _get_make_arguments()
```

Comment:
The `--` has nothing to do with the make arguments but with the fact that the invoked cmd is `cmake --build` and arguments which should be passed along to the specific tool need to be separated by a double dash. Therefore I will remove this comment.